OrbitGl/ScopedStatus.h

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_GL_SCOPED_STATUS_H_
+#define ORBIT_GL_SCOPED_STATUS_H_
+
+#include <string>
+#include <thread>
+
+#include "MainThreadExecutor.h"
+#include "OrbitBase/Logging.h"
+#include "StatusListener.h"
+
+/**
+ * This class holds scope for status. It also takes care
+ * of updating status on the main thread even if Update
+ * is called from a different thread.
+ *
+ * Usage example:
+ *
+ * {
+ *   ScopedStatus scoped_status =
+ *       new ScopedStatus(main_thread_executor, status_listener_, "Waiting for fish");
+ *
+ *   ...
+ *
+ *   scoped_status.Update("Found fish!");
```

Comment:
Nit: "Found fish" somehow gives the impression that the `Update` message marks the end of the status display. But that's not true, is it?

But it does what it says. It updates the message and  continues to display it.

Reply:
done.